Fix horizontal navigation menu example

maspitz opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
<!-- ✍️ In a sentence or two, describe your changes -->
Removes the spaces and line breaks between the &lt;li&gt; elements
in the horizontal nav menu example.

#### Motivation
<!-- ❓ Why are you making this change? Help us understand how your changes help readers. -->
As noted in the the tutorial's text, there must be no spaces or line spaces
between the &lt;li&gt; elements.

Otherwise, the fifth item ("Finland") ends up being placed
on a line by itself, contrary to the intended nav menu design.

#### Supporting details
<!-- 🔗 Link to supporting information, such as bug trackers, source control, release notes, and vendor docs. -->
https://css-tricks.com/fighting-the-space-between-inline-block-elements/

#### Related issues
<!-- 🔨 If applicable, use "Fixes #XYZ" -->

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x ] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

maspitz committed:
Fix horizontal navigation menu example

Removes the spaces and line breaks between the <li> elements
in the horizontal nav menu example.

As noted in the the tutorial's text, there must be no spaces or line spaces
between the <li> elements.

Otherwise, the fifth item ("Finland") ends up being placed
on a line by itself, contrary to the intended nav menu design.

OnkarRuikar commented on files/en-us/learn/css/styling_text/styling_links/index.md:
```diff
@@ -387,11 +387,7 @@ First, some HTML:
-  <li><a href="#">Music</a></li>
-  <li><a href="#">Wombats</a></li>
-  <li><a href="#">Finland</a></li>
+  <li><a href="#">Home</a></li><li><a href="#">Pizza</a></li><li><a href="#">Music</a></li><li><a href="#">Wombats</a></li><li><a href="#">Finland</a></li>
```
I think the comment fix would be more suitable/presentable for the demo. And it'll also shine more light on the issue. 
```suggestion
  <li><a href="#">Home</a></li><!--
  --><li><a href="#">Pizza</a></li><!--
  --><li><a href="#">Music</a></li><!--
  --><li><a href="#">Wombats</a></li><!--
  --><li><a href="#">Finland</a></li>
```

maspitz committed:
Fix links-as-buttons example using comment method

Co-authored-by: Onkar Ruikar <87750369+OnkarRuikar@users.noreply.github.com>

teoli2003 commented on files/en-us/learn/css/styling_text/styling_links/index.md:
````diff
@@ -387,7 +387,11 @@ First, some HTML:
 ```html
 <ul>
-  <li><a href="#">Home</a></li><li><a href="#">Pizza</a></li><li><a href="#">Music</a></li><li><a href="#">Wombats</a></li><li><a href="#">Finland</a></li>
+  <li><a href="#">Home</a></li><!--
+  --><li><a href="#">Pizza</a></li><!--
+  --><li><a href="#">Music</a></li><!--
+  --><li><a href="#">Wombats</a></li><!--
+  --><li><a href="#">Finland</a></li>
````
I think this is the more legible.

```suggestion
  <li><a href="#">Home</a></li>
  <li><a href="#">Pizza</a></li>
  <li><a href="#">Music</a></li>
  <li><a href="#">Wombats</a></li>
  <li><a href="#">Finland</a></li>
```

teoli2003 replied:
Oh yes, you are right, I remember. I complained about it once. So I think we should keep it horizontal then.

maspitz replied:
Just so I understand clearly, by keeping it horizontal, do you mean, you prefer the method of the single line `<li><a href="#">Home</a></li><li><a href="#">Pizza</a></li><li><a href="#">Music</a></li><li><a href="#">Wombats</a></li><li><a href="#">Finland</a></li>` without comment-protected newlines?  Thanks.

teoli2003 replied:
Yes, I do. If we add comment-protected newlines, we should explain why and I think this is out-of-scope for this page.

maspitz replied:
Ok, it is the simplest way, and the example is already relatively complicated.

```suggestion
  <li><a href="#">Home</a></li><li><a href="#">Pizza</a></li><li><a href="#">Music</a></li><li><a href="#">Wombats</a></li><li><a href="#">Finland</a></li>
```

maspitz committed:
Revise note regarding inline block element spacing

This updates the note on the "styling links as buttons" example
to explain the solution of keeping the list items on separate lines,
using comments to contain the spacing.

maspitz committed:
Merge branch 'mdn:main' into patch-1

maspitz committed:
Revise note regarding inline block element spacing

This reverts the note on the "styling links as buttons" example
to explain the solution of keeping the list items on a single lines.

It restores the previous text, and removes the references to using
the alternative strategy of using comments in the HTML to contain
the spacing.

maspitz committed:
Simplify links-as-buttons example to use single line

Rather than using comments to split lines, just keep the
list items on a single line in the example's HTML.

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

teoli2003 reviewed (approved):
👍

teoli2003 commented:
Thanks a lot!

Looks good to me.

Congratulations upon your first merged commit here! Welcome aboard! :tada:
